extendr-api/src/scalar/macros.rs

```diff
@@ -743,7 +766,7 @@ macro_rules! gen_trait_impl {
             impl PartialEq<$type_prim> for $type {
                 /// NA always fails.
                 fn eq(&self, other: &$type_prim) -> bool {
-                    !self.is_na() && self.0 == *other
+                    <Option<$type_prim>>::try_from(self.clone()) == Ok(Some(*other))
```

Comment:
While I can't find any combination of the inputs that produce an incorrect result, I feel like `Some(*other)` is not the best choice here. What if `other` contains a value that is `NA`? Then `Some(NA)` does not actually make sense, it should be `None`. Is it designed so only to ensure that `NA` always fails? Wouldn't it be better for maintainability to explicitly check for `NA` and then perform an equality check on `Result<Option<$type_prim>, ()>`? For that we need something like `impl TryFrom<Option<$type_prim>> for $type_prim`  though, and I am not sure it is a good idea.

Reply:
If both are NA then try_from will produce Err() and the test will fail.

If NA comparisons always fail, this is correct.

Reply:
Ok, fair enough.